daemon/cluster/listen_addr.go

```diff
@@ -7,12 +7,13 @@ import (
 )
 
 var (
-	errNoSuchInterface         = errors.New("no such interface")
-	errNoIP                    = errors.New("could not find the system's IP address")
-	errMustSpecifyListenAddr   = errors.New("must specify a listening address because the address to advertise is not recognized as a system address, and a system's IP address to use could not be uniquely identified")
-	errBadListenAddr           = errors.New("listen address must be an IP address or network interface (with optional port number)")
-	errBadAdvertiseAddr        = errors.New("advertise address must be an IP address or network interface (with optional port number)")
-	errBadDefaultAdvertiseAddr = errors.New("default advertise address must be an IP address or network interface (without a port number)")
+	errNoSuchInterface          = errors.New("no such interface")
+	errNoIP                     = errors.New("could not find the system's IP address")
+	errMustSpecifyListenAddr    = errors.New("must specify a listening address because the address to advertise is not recognized as a system address, and a system's IP address to use could not be uniquely identified")
+	errBadListenAddr            = errors.New("listen address must be an IP address or network interface (with optional port number)")
+	errBadAdvertiseAddr         = errors.New("advertise address must be an IP address or network interface (with optional port number)")
+	errBadDefaultAdvertiseAddr  = errors.New("default advertise address must be an IP address or network interface (without a port number)")
+	errUnspecifiedAdvertiseAddr = errors.New("the address to advertise must not be an unspecified (all zeros) IP address")
```

Comment:
Not in this PR right? I'd rather not have to go hunting for all of the places which use or check for certain error string values, etc. I suppose we'd also have to make these changes in `docker/swarmkit`. For consistency then, I'll just change this this to `"the advertise address ..."`.